File: app/src/main/java/com/intoverflown/pos/ui/inventory/InventoryActivityMain.java
package com.intoverflown.pos.ui.inventory;

import android.content.Intent;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import com.intoverflown.pos.databinding.ActivityInventoryMainBinding;
import com.intoverflown.pos.ui.inventory.addproduct.AddProductActivity;

public class InventoryActivityMain extends AppCompatActivity {

    private ActivityInventoryMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        binding = ActivityInventoryMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        binding.fab.setOnClickListener(v -> {
            Intent i = new Intent(InventoryActivityMain.this, AddProductActivity.class);
            i.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
            startActivity(i);
        });
    }
}

File: app/src/main/java/com/intoverflown/pos/ui/inventory/InventoryActivityMain.kt
package com.intoverflown.pos.ui.inventory

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.intoverflown.pos.databinding.ActivityInventoryMainBinding
import com.intoverflown.pos.ui.inventory.addproduct.AddProductActivity

class InventoryActivityMain : AppCompatActivity() {

    private var binding : ActivityInventoryMainBinding? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityInventoryMainBinding.inflate(layoutInflater)
        setContentView(binding!!.root)

        binding!!.fab.setOnClickListener {
            val i = Intent(this, AddProductActivity::class.java)
            i.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
            startActivity(i)
        }
    }
}